js/Scene3D.js

```diff
@@ -1,399 +1,408 @@
+                    this._recolor();
+                    this._notify(Scene3D.Events.CHANGE);
+                }
+            }
         },
 
-        set: function(value) {
-            this._colorMap = value;
-            if (this._mesh && this._spots && this._mapping) {
-                this._recolor();
-                this._notify(Scene3D.Events.CHANGE);
+        position: {
+            get: function () {
+                return this._scene.position.clone();
             }
-        }
-    },
+        },
 
-    position: {
-        get: function() {
-            return this._scene.position.clone();
-        }
-    },
+        render: {
+            value: function (renderer, camera) {
+                this._frontLight.position.set(camera.position.x, camera.position.y, camera.position.z);
+                renderer.render(this._scene, camera);
+            }
+        },
 
-    render: {
-        value: function(renderer, camera) {
-            this._frontLight.position.set(camera.position.x, camera.position.y, camera.position.z);
-            renderer.render(this._scene, camera);
-        }
-    },
-
-    raycast: {
-        value: function(raycaster) {
-            if (!this._mesh || !this._spots || !this._mapping) return null;
-            var message = {
-                positions: this._mesh.geometry.attributes.position.array,
-                origin: new THREE.Vector3().copy(raycaster.ray.origin),
-                direction: new THREE.Vector3().copy(raycaster.ray.direction),
-                matrixWorld: new THREE.Matrix4().copy(this._mesh.matrixWorld),
-            };
-            var closestSpotIndeces = this._mapping.closestSpotIndeces;
-            var spots = this._spots;
-            var worker = new Worker('js/workers/Raycaster.js');
-
-            var promise = new Promise(function(accept, reject) {
-                worker.onmessage = function(event) {
-                    worker.terminate();
-                    var face = event.data;
-                    var spotIndex = -1;
-                    for (var i in (face || {})) {
-                        if (closestSpotIndeces[face[i]] >= 0) {
-                            spotIndex = closestSpotIndeces[face[i]];
-                            break;
+        raycast: {
+            value: function (raycaster) {
+                if (!this._mesh || !this._spots || !this._mapping) return null;
+                var message = {
+                    positions: this._mesh.geometry.attributes.position.array,
+                    origin: new THREE.Vector3().copy(raycaster.ray.origin),
+                    direction: new THREE.Vector3().copy(raycaster.ray.direction),
+                    matrixWorld: new THREE.Matrix4().copy(this._mesh.matrixWorld),
+                };
+                var closestSpotIndeces = this._mapping.closestSpotIndeces;
+                var spots = this._spots;
+                var worker = new Worker('js/workers/Raycaster.js');
```

Comment:
@ElDeveloper, I also saw this example with a unit-test, but what confused me about it is that they call `require` for that worker rather than `define`. As far as I understand, `require` means you want to load a piece of of code with all its dependencies right away, whereas with `define`, you create a module that can be used later. The latter option seems closer to how workers are normally used. Am I missing something?